test.disabled/ejabberd_tests/tests/mam_SUITE.erl

```diff
@@ -2538,6 +2571,191 @@ pagination_last_after_id5_before_id11(Config) ->
+        RSM = #rsm_in{max=5, direction=before, id=message_id(6, Config)},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"before6">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"true">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% First page is not compete, because max is smaller than archive size.
+%% rsm_id.id is undefined
+%% GIVEN 15 archived messages
+%% WHEN direction=after, rsm_id=6, page_size=5
+%% THAN complete=false
+after_complete_false_first_page(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get messages: 1,2,3,4,5
+        RSM = #rsm_in{max=5, direction='after'},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"firstpage">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"false">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% There are still 8-15 messages to paginate after this request.
+%% GIVEN 15 archived messages
+%% WHEN direction=after, rsm_id=2, page_size=5
+%% THAN complete=false
+after_complete_false_after2(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get messages: 3,4,5,6,7
+        RSM = #rsm_in{max=5, direction='after', id=message_id(2, Config)},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"after2">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"false">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% There is still one message to paginate after this request.
+%% GIVEN 15 archived messages
+%% WHEN direction=after, rsm_id=9, page_size=5
+%% THAN complete=false
+after_complete_false_after9(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get messages: 10,11,12,13,14
+        RSM = #rsm_in{max=5, direction='after', id=message_id(9, Config)},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"after9">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"false">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% There are no messages to paginate after this request.
+%% Special case, when exactly page_size messages are returned.
+%% GIVEN 15 archived messages
+%% WHEN direction=after, rsm_id=10, page_size=5
+%% THAN complete=true
+after_complete_true_after10(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get the last page of size 5.
+        %% Get messages: 11,12,13,14,15
+        RSM = #rsm_in{max=5, direction='after', id=message_id(10, Config)},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"after10">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"true">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% There are no messages to paginate after this request.
+%% Less than page_size are returned.
+%% GIVEN 15 archived messages
+%% WHEN direction=after, rsm_id=10, page_size=5
+%% THAN complete=true
```

Comment:
`THAN` => `THEN` in all comments in new test cases.

Reply:
yeah

Reply:
Why?